markets/perps-market/contracts/modules/AsyncOrderSettlementPythModule.sol

```diff
@@ -113,26 +131,22 @@ contract AsyncOrderSettlementPythModule is
 
         runtime.settlementReward = AsyncOrder.settlementRewardCost(settlementStrategy);
 
+        // if settlement reward is non-zero, pay keeper
         if (runtime.settlementReward > 0) {
-            // pay keeper
             factory.withdrawMarketUsd(ERC2771Context._msgSender(), runtime.settlementReward);
         }
 
-        (runtime.referralFees, runtime.feeCollectorFees) = GlobalPerpsMarketConfiguration
-            .load()
-            .collectFees(
-                runtime.totalFees - runtime.settlementReward, // totalFees includes settlement reward so we remove it
-                asyncOrder.request.referrer,
-                factory
-            );
+        // order fees are total fees minus settlement reward
+        uint orderFees = runtime.totalFees - runtime.settlementReward;
+        GlobalPerpsMarketConfiguration.Data s = GlobalPerpsMarketConfiguration.load();
+        s.collectFees(orderFees, asyncOrder.request.referrer, factory);
 
         // trader can now commit a new order
         asyncOrder.reset();
 
-        // Note: new event for this due to stack too deep adding it to OrderSettled event
+        /// @custom:question two events here seems bad. Can we not combine or find a better way?
```

Comment:
combining the two events might be possible. we can take a look at this next week and sync up with the data and UI teams to see what sort of data they are using from this event in the data repo or clients and make sure they are aware of any changes so they can update their code as well

Reply:
You want to avoid stack too deep issues, hence 2 events

Reply:
yes, but the above stack limitation can be gracefully circumvented without need of two events (bad code).

From [soliditylang abi specs for events](https://docs.soliditylang.org/en/latest/abi-spec.html#events):

> Events are an abstraction of the Ethereum logging/event-watching protocol. Log entries provide the contract’s address, a series of up to four topics and some arbitrary length binary data. 

Given: "arbitrary length binary data"
We can say: Any number of variables can be encoded and thus emitted without concern of the stack

Reply:
this shouldnt be changed here regardless but it is a code quality issue that should be logged... somewhere.....? @kaleb-keny 😅

Reply:
Yeah i think sunny might have mentioned this to me once long time back, that splitting events was required for avoiding that stack-too-deep error.